File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-ios-lib/AbtTokenCoreSDK/keystore/DefaultKeyStoreAliasFactory.swift
import Foundation

public class DefaultKeyStoreAliasFactory: KeyStoreAliasFactory {
    private enum K {
        static let separator = ":" // do not use '-' as it is in the token
        static let encryptationKey = "encryption"
        static let signatureKey = "token"
        static let defaultKeyAliasPrefix = "entur-tokens"
        static let certificateKey = "certificate"
    }

    private let globalPrefixWithPrefix: String

    public init() {
        globalPrefixWithPrefix = K.defaultKeyAliasPrefix + K.separator
    }

    public init(globalPrefixWithPrefix prefix: String) {
        globalPrefixWithPrefix = prefix + K.separator
    }

    public func getAlias(tokenContext: TokenContext, tokenId: String, alias: String) -> String {
        getTokenAliasPrefix(tokenContext: tokenContext, tokenId: tokenId) + alias
    }

    public func getTokenAliasPrefix(tokenContext: TokenContext, tokenId: String) -> String {
        getContextAliasPrefix(tokenContext: tokenContext) + tokenId + K.separator
    }

    public func getContextAliasPrefix(tokenContext: TokenContext) -> String {
        globalPrefixWithPrefix + tokenContext.id + K.separator
    }

    public func getGlobalAliasPrefix() -> String {
        globalPrefixWithPrefix
    }

    public func getEncryptionKeyAlias(tokenContext: TokenContext, tokenId: String) -> String {
        getAlias(tokenContext: tokenContext, tokenId: tokenId, alias: K.encryptationKey)
    }

    public func getSignatureKeyAlias(tokenContext: TokenContext, tokenId: String) -> String {
        getAlias(tokenContext: tokenContext, tokenId: tokenId, alias: K.signatureKey)
    }

    public func getSignatureCertificateKeyAlias(tokenContext: TokenContext, tokenId: String) -> String {
        getAlias(tokenContext: tokenContext, tokenId: tokenId, alias: K.signatureKey + K.separator + K.certificateKey)
    }
}

File: .yalc/@entur/atb-mobile-client-sdk/token/token-core-android-lib/src/main/java/no/entur/abt/android/token/keystore/DefaultKeystoreAliasFactory.java
package no.entur.abt.android.token.keystore;

public class DefaultKeystoreAliasFactory implements KeystoreAliasFactory {

	protected static final char SEPERATOR = ':'; // do not use '-' as it is in the token

	protected static final String ENCRYPTION_KEY = "encryption";
	protected static final String SIGNATURE_KEY = "token";

	protected final String globalPrefixWithPrefix;

	public DefaultKeystoreAliasFactory(String globalPrefix) {
		this.globalPrefixWithPrefix = globalPrefix + SEPERATOR;
	}

	@Override
	public String getEncryptionKeyAlias(String tokenContextId, String tokenId) {
		return getAlias(tokenContextId, tokenId, ENCRYPTION_KEY);
	}

	@Override
	public String getSignatureKeyAlias(String tokenContextId, String tokenId) {
		return getAlias(tokenContextId, tokenId, SIGNATURE_KEY);
	}

	@Override
	public String getAlias(String tokenContextId, String tokenId, String alias) {
		return getTokenAliasPrefix(tokenContextId, tokenId) + alias;
	}

	public String getTokenAliasPrefix(String tokenContextId, String tokenId) {
		return getContextAliasPrefix(tokenContextId) + tokenId + SEPERATOR;
	}

	public String getContextAliasPrefix(String tokenContextId) {
		return globalPrefixWithPrefix + tokenContextId + SEPERATOR;
	}

	public String getGlobalAliasPrefix() {
		return globalPrefixWithPrefix;
	}

}
